internal/pkg/deploy/pipeline.go

```diff
@@ -637,6 +751,33 @@ func (stg *PipelineStage) buildDeploymentsGraph() *graph.Graph[string] {
 	return digraph
 }
 
+func (stg *PipelineStage) buildPrePostDeploymentsGraph(depType string) *graph.Graph[string] {
```

Comment:
I feel like we can reuse https://github.com/aws/copilot-cli/blob/e4081a0248ec1a8e302f15d144f3effc6ca45a77/internal/pkg/deploy/pipeline.go#L620 with some refactor! For example,
```go
type rankable interface{
  dependsOn()
  name()
}
func (stg *PipelineStage) buildDependencyGraph(rankable []rankable) *graph.Graph[string] {}

// Call this function with
stg.buildDependencyGraph(stg.deployments)
stg.buildDependencyGraph(stg.preDeployments)
stg.buildDependencyGraph(stg.postDeployments)

```

Reply:
Ah, this is a much better way than the many ways I tried to do this. 😆